File: Practica3/Cola/src/estructura1/Cola.java
package estructura1;

public class Cola {
    private long[] arreglo;
    private int inicio;
    private int fin;
    private int n;

    public Cola(int n) {
        this.n = n;
        arreglo = new long[n];
        inicio = 0;
        fin = -1;
    }

    public void insert(long e) {
        if (!isFull()) {
            arreglo[++fin] = e;
        } else {
            System.out.println("Cola llena");
        }
    }

    public long remove() {
        if (!isEmpty()) {
            return arreglo[inicio++];
        }
        System.out.println("Cola vacía");
        return -1;
    }

    public long peek() {
        if (!isEmpty()) {
            return arreglo[inicio];
        }
        System.out.println("Cola vacía");
        return -1;
    }

    public boolean isEmpty() {
        return inicio > fin;
    }

    public boolean isFull() {
        return fin == n - 1;
    }

    public int size() {
        return fin - inicio + 1;
    }

    public static void main(String[] args) {
        Cola cola = new Cola(3);
        
        cola.insert(10);
        cola.insert(20);
        cola.insert(30);

        System.out.println("Frente de la cola: " + cola.peek());
        System.out.println("Tamaño de la cola: " + cola.size());

        System.out.println("Eliminando: " + cola.remove());
        System.out.println("Nuevo frente de la cola: " + cola.peek());
        System.out.println("Tamaño de la cola: " + cola.size());
    }
}

File: Practica3/Cola.py
class Cola:
    def __init__(self, n):
        self.n = n
        self.arreglo = [-1] * n  
        self.inicio = 0
        self.fin = -1

    def insert(self, e):
        if not self.isFull():
            self.fin += 1
            self.arreglo[self.fin] = e
        else:
            print("Cola llena!")

    def remove(self):
        if not self.isEmpty():
            valor = self.arreglo[self.inicio]
            self.inicio += 1
            return valor
        return -1  

    def peek(self):
        if not self.isEmpty():
            return self.arreglo[self.inicio]
        return -1  

    def isEmpty(self):
        return self.inicio > self.fin

    def isFull(self):
        return self.fin == self.n - 1

    def size(self):
        return self.fin - self.inicio + 1

if __name__ == "__main__":
    cola = Cola(3)
    
    cola.insert(10)
    cola.insert(20)
    cola.insert(30)
    cola.insert(40)

    print("Frente de la cola:", cola.peek())
    print("Tamaño de la cola:", cola.size())
